draft-ietf-quic-invariants.md

```diff
@@ -178,16 +191,17 @@ version-specific semantics are marked with an X.
 
 A QUIC packet with a short header has the high bit of the first octet set to 0.
 
-A QUIC packet with a short header includes an optional connection ID and no
-version field.  The second bit of that octet (that is, 0x40) determines whether
-the connection ID is present.  If the second bit is cleared, a 64-bit connection
-ID immediately follows the first octet.  If the second bit is set, the remainder
-of the packet has version-specific semantics.
+A QUIC packet with a short header includes a Destination Connection ID.  The
+short header does not include the Connection ID Lengths, Source Connection ID,
+or Version fields.
+
+The remainder of the packet has version-specific semantics.
 
 
 ## Connection ID
 
-A connection ID is an opaque 64-bit field.
+A connection ID is an opaque field.  A connection ID can be 0 octets in length,
+or between 8 and 22 octets (inclusive).
```

Comment:
Why limit the allowable range to 8 and 22 (inclusively)? Practically, the size just needs to be large enough for the number of connections on a given tuple, right? So if a client only needs to share with a couple of connections, 1 byte might be sufficient.

Reply:
I thought carefully about this.

Shorter values make it much harder to come up with a scheme for preserving unlinkability if you use NEW_CONNECTION_ID.  Thus, I want to prohibit 1 octet connection IDs.  2 seems similarly challenged.